pytorch_lightning/trainer/trainer.py

```diff
@@ -685,14 +685,13 @@ def _call_and_handle_interrupt(self, trainer_fn: Callable, *args: Any, **kwargs:
         except BaseException as exception:
             self.state.status = TrainerStatus.INTERRUPTED
             if distributed_available() and self.world_size > 1:
-                # try syncing remaing processes, kill otherwise
+                # try syncing remaining processes, kill otherwise
                 self.strategy.reconciliate_processes(traceback.format_exc())
             self._on_exception()
+            self._call_callback_hooks("on_exception", exception)
+            self._teardown()
             # reset bookkeeping
             self.state.stage = None
```

Comment:
The stage needs to be the last thing to unset as `teardown` might access it

Reply:
Maybe we should add a comment for that to prevent that from changing again?